marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/transforms/generate-function-metadata.sjs

```diff
@@ -21,9 +21,17 @@ function transform(context, params, content) {
       ])`;
       let writeInfo = datahub.hubUtils.writeDocument(uriVal + ".xml", metaDataXml, permissionsExpression, [collection], datahub.config.MODULESDATABASE);
       if (writeInfo && fn.exists(writeInfo.transaction)) {
-        // Workaround to avoid XSLT-UNBPRFX error. See https://bugtrack.marklogic.com/52870
-        xdmp.moduleCacheClear();
-        es.functionMetadataPut(uriVal + ".xml");
+        // try/catch workaround to avoid XSLT-UNBPRFX error. See https://bugtrack.marklogic.com/52870
+        try {
+          es.functionMetadataPut(uriVal + ".xml");
+        } catch (e) {
+          if (/(prefix|XSLT-UNBPRFX)/ig.test(e.message)) {
```

Comment:
I think it's worth adding a small unit test to verify this regular expression, with the check being pulled into a separate function. That function (and the test) become better places to document what's going on here as well, rather than an inline comment. 

Seems like it's worth veryfing that the result of "functionMetadataPut" still occurs as well.